OpenSim/Simulation/SimbodyEngine/SpatialTransform.cpp

```diff
@@ -193,7 +193,9 @@ void SpatialTransform::scale(const SimTK::Vec3 scaleFactors)
             }
             SimTK::Vec3 axis;
             transform.getAxis(axis);
-            double scaleFactor = ~axis * scaleFactors;
+            // we want weighted aggregate of scale factors but to ignore the sign
+            // ignoring sign due to issue #3991 resulting -ve scale factor
+            double scaleFactor = ~axis.abs() * scaleFactors;
```

Comment:
Seems like a reasonable enough change; but it makes me wonder: is there any scenario where we would want to preserve the sign on the scale factor?

Reply:
While manual scales can be set to -ve numbers, measurement-based scale factors are ratios between (unsigned) distances so can't be -ve. If we think about it as how much we stretch/shrink segments then this should never be -ve otherwise vectors/directions flip which is unlikely an intended behavior of our 'scaling'. However if you suspect there're other places in the code worth more investigation please let me know @nickbianco

Reply:
Sounds good, just making sure I understood this correctly.